### DISCUSSION TITLE
Custom sampler using javaagent

### DISCUSSION THREAD

**Discussion by bgong-mdsol** [open-telemetry/opentelemetry-java-instrumentation#6089] [Q&A]
I am using java agent with a custom sampler which can decide sampling the `Internal` span or not based on a condition, see below:

```java
public class MySampler implements Sampler {

final static boolean isDisableInternalSpan = disableInternalSpan();
 ...

  @Override
  public SamplingResult shouldSample(Context parentContext, String traceId, String name,
      SpanKind spanKind, Attributes attributes, List<LinkData> parentLinks) {
    SpanContext parentSpanContext = Span.fromContext(parentContext).getSpanContext();
    if (spanKind.equals(SpanKind.INTERNAL) && isDisableInternalSpan) {
      return SamplingResult.create(SamplingDecision.DROP);
    } else {
      return traceSampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
    }
  }

...

}
```

It will drop the  the `Internal` span and its child spans, is it possible to drop `Internal` span only, e.g
The trace tree: `Server span` -> `internal span` -> `DB span`
The expected tree without internal span: `Server span` -> `DB span`

**Comment by laurit** [comment]
Perhaps the `traceSampler` you delegate to is a parent based sampler that samples only when parent span was sampled?

**Reply by trask** [reply]
If you use a non-parent based sampler, you could capture the db span, but I don't think it would be parented to the server span anymore.

If you are looking to suppress INTERNAL spans produced by the standard javaagent instrumentation, most of those are suppressible using https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#suppressing-controller-andor-view-spans
